### DISCUSSION TITLE
Hint text in tuiAsideItem calculated in wrong moment (Angular 19)

### DISCUSSION THREAD

**Discussion by mokeev1995** [taiga-family/taiga-ui#10107] [Q&A] [answered]
Let's imagine we need to do some badge near title of `tuiAsideItem` (using Navigation components set) only in expanded state

![image](https://github.com/user-attachments/assets/36c3bb72-bf28-4696-944b-babde29dc73e)

So I placed the badge inside that `tuiAsideItem` and hide it with basic `@if (expanded()) {}`

<details>
  <summary>Example</summary>

  ![image](https://github.com/user-attachments/assets/6b925fed-1a8e-4bbd-bcc1-1d5596ecd606)
</details>

<details>
  <summary>Template content</summary>

  ![image](https://github.com/user-attachments/assets/65016cad-212d-4b94-82dd-ad985f1f6ca2)
</details>

After that I got that sort of a hint on element in collapsed state:

![image](https://github.com/user-attachments/assets/a49b3ed4-3cf2-4a36-8e5d-07d2b183e2bb)

So, as I discovered, this code executes exactly after the collapse was triggered, before `@if (...)` even rerendered.

![image](https://github.com/user-attachments/assets/8e76f1c5-4980-49cc-80cc-0f5565359a83)

How am I supposed to write that to solve the issue with wrong title in the hint?

Here's my package.json

<details>
  <summary>package.json</summary>

```json
{
  "dependencies": {
    "@angular/animations": "^19.0.0",
    "@angular/cdk": "^19.0.0",
    "@angular/common": "^19.0.0",
    "@angular/compiler": "^19.0.0",
    "@angular/core": "^19.0.0",
    "@angular/forms": "^19.0.0",
    "@angular/platform-browser": "^19.0.0",
    "@angular/platform-browser-dynamic": "^19.0.0",
    "@angular/platform-server": "^19.0.0",
    "@angular/router": "^19.0.0",
    "@angular/ssr": "^19.0.6",
    "@ng-web-apis/universal": "^4.11.1",
    "@taiga-ui/addon-charts": "^4.19.0",
    "@taiga-ui/addon-commerce": "^4.19.0",
    "@taiga-ui/addon-table": "^4.19.0",
    "@taiga-ui/cdk": "^4.19.0",
    "@taiga-ui/core": "^4.19.0",
    "@taiga-ui/event-plugins": "^4.0.2",
    "@taiga-ui/icons": "^4.19.0",
    "@taiga-ui/kit": "^4.19.0",
    "@taiga-ui/layout": "^4.19.0",
    "eslint-plugin-unused-imports": "^4.1.4",
    "express": "^4.18.2",
    "rxjs": "~7.8.1",
    "tslib": "^2.8.1",
    "zone.js": "~0.15.0"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^19.0.6",
    "@angular/cli": "^19.0.6",
    "@angular/compiler-cli": "^19.0.0",
    "@stylistic/eslint-plugin": "^2.12.1",
    "@types/express": "^4.17.17",
    "@types/jasmine": "~5.1.0",
    "@types/node": "^22.10.2",
    "angular-eslint": "19.0.2",
    "eslint": "^9.16.0",
    "jasmine-core": "~5.4.0",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.2.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.1.0",
    "typescript": "~5.6.2",
    "typescript-eslint": "8.18.0"
  }
}
```
</details>

TLDR: angular 19, taiga 4.19

Here's the repro: [https://stackblitz.com/edit/angular-g3eos13a](https://stackblitz.com/edit/angular-g3eos13a?file=src%2Fapp%2Fapp.component.ts)

![image](https://github.com/user-attachments/assets/ea616f10-4b38-4b82-ac91-a6ae3eea8069)

**Comment by waterplea** [comment]
Might be an issue on our side, it should probably recalculate every time you hover

**Reply by mokeev1995** [reply]
FYI: It seems it works fine with angular 18, something has changed in 19 that brought us this new behavior

**Reply by waterplea** [reply]
Interesting. Can you file an issue mentioning that?

**Reply by mokeev1995** [reply, marked as answer]
Done https://github.com/taiga-family/taiga-ui/issues/10111
